src/blockchain_db/sqlite/db_sqlite.cpp

```diff
@@ -305,28 +390,67 @@ bool BlockchainSQLite::pop_block(const cryptonote::block &block, std::vector<cry
-    SQLite::Statement delete_payment{db,
-      "DELETE from batched_payments WHERE height_earned = ?"};
-    db::exec_query(delete_payment, static_cast<int64_t>(block_height));
-
-    // Marks the miner tx payments that received funds in this block as unpaid (paid_height = NULL)
+    // Add back to the database payments that had been made in this block
     delete_block_payments(block_height);
 
+    // Step 1: Remove the block producers txn fees
+    uint64_t service_node_reward = cryptonote::service_node_reward_formula(0, block.major_version);
+    if (block.reward - service_node_reward > 0)
+    {
+      if (block.service_node_winner_key && crypto_core_ed25519_is_valid_point(reinterpret_cast<const unsigned char *>(block.service_node_winner_key.data)))
+      {
+        auto service_node_winner = service_nodes_state.service_nodes_infos.find(block.service_node_winner_key);
+        if (service_node_winner != service_nodes_state.service_nodes_infos.end())
+        {
+          std::vector<cryptonote::batch_sn_payment> block_producer_fee_payments = calculate_rewards(block.major_version, block.reward - service_node_reward, *service_node_winner->second);
+          // Takes the block producer and adds its contributors to the batching database for the transaction fees
+          success = subtract_sn_payments(block_producer_fee_payments);
+          if (!success)
+            return false;
+          }
+        }
+    }
+
+    // Step 2: Iterate over the whole service node list and subtract each node 1/service_node_list fraction
+    const auto payable_service_nodes = service_nodes_state.payable_service_nodes_infos(block_height);
+    size_t total_service_nodes_payable = payable_service_nodes.size();
+    for (const auto& [node_pubkey, node_info]: payable_service_nodes)
+    {
+      std::vector<cryptonote::batch_sn_payment> node_contributors;
+      auto payable_service_node = service_nodes_state.service_nodes_infos.find(node_pubkey);
+      if (payable_service_node == service_nodes_state.service_nodes_infos.end())
+        continue;
+      std::vector<cryptonote::batch_sn_payment> node_rewards = calculate_rewards(block.major_version, service_node_reward/total_service_nodes_payable, *payable_service_node->second);
+      // Takes the node and adds its contributors to the batching database
+      success = subtract_sn_payments(node_rewards);
+      if (!success)
+        return false;
+    }
+    // Step 3: Remove Governance reward
+    if (m_nettype != cryptonote::FAKECHAIN)
+    {
+      std::vector<cryptonote::batch_sn_payment> governance_rewards;
+      cryptonote::address_parse_info governance_wallet_address;
+      cryptonote::get_account_address_from_str(governance_wallet_address, m_nettype, cryptonote::get_config(m_nettype).governance_wallet_address(hf_version));
+      governance_rewards.emplace_back(governance_wallet_address.address, FOUNDATION_REWARD_HF17, m_nettype);
```

Comment:
this should be fetching foundation reward based on hf_version just as the line above is doing for the address, right?